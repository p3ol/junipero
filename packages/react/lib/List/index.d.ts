import { ReactNode, ComponentPropsWithRef, MutableRefObject } from 'react';

export declare type ListRef = {
  orderable: boolean;
  columns: Array<any>;
  active: string | number;
  asc: boolean | null;
  isJunipero: boolean;
  innerRef: MutableRefObject<any>;
};

declare interface ListProps extends ComponentPropsWithRef<any> {
  columns?: Array<any>;
  onOrder?(order: { column: string | number, asc: boolean | null }): void;
  ref?: MutableRefObject<ListRef | undefined>;
}

declare function List(props: ListProps): ReactNode | JSX.Element;

export default List;
