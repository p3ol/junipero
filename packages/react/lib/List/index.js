import {
  forwardRef,
  useImperativeHandle,
  useRef,
  useReducer,
  useCallback,
  useMemo,
} from 'react';
import { mockState, classNames } from '@junipero/core';
import PropTypes from 'prop-types';

import { ListContext } from '../contexts';
import { ArrowDown, ArrowUp } from '../icons';

const List = forwardRef(({
  className,
  children,
  columns = [],
  onOrder,
  ...rest
}, ref) => {
  const listRef = useRef([]);
  const innerRef = useRef();
  const orderable = useMemo(() => !!onOrder, [onOrder]);
  const [state, dispatch] = useReducer(mockState, {
    columns,
    active: null,
    asc: null,
  });

  useImperativeHandle(ref, () => ({
    innerRef,
    orderable,
    columns: state.columns,
    active: state.active,
    asc: state.asc,
    isJunipero: true,
  }));

  const onOrder_ = (column, e) => {
    e?.preventDefault();

    const asc = state.asc === true ? null : state.asc === false;

    dispatch({ active: column, asc });
    onOrder?.({ column, asc });
  };

  const registerColumn = column => {
    if (listRef.current.find(it => it.id === column.id)) {
      return;
    }

    listRef.current.push(column);
    dispatch({ columns: listRef.current });
  };

  const getContext = useCallback(() => ({
    active: state.active,
    asc: state.asc,
    orderable,
    registerColumn,
  }), [
    state.columns,
    state.active,
    state.asc,
    orderable,
  ]);

  const renderColumn = (column, index) => {
    const { id, title, ...props } = typeof column === 'string'
      ? { id: column, title: column } : column;

    return (
      <th {...props} key={id ?? index}>
        { orderable ? (
          <a href="#" onClick={onOrder_.bind(null, id)}>
            <span>{ title }</span>
            { state.active === id && state.asc === true ? (
              <ArrowUp />
            ) : state.active === id && state.asc === false ? (
              <ArrowDown />
            ) : null }
          </a>
        ) : title }
      </th>
    );
  };

  return (
    <ListContext.Provider value={getContext()}>
      <table
        {...rest}
        className={classNames('junipero', 'list', className)}
        ref={innerRef}
      >
        <thead>
          <tr>{ state.columns.map(renderColumn) }</tr>
        </thead>
        <tbody>
          { children }
        </tbody>
      </table>
    </ListContext.Provider>
  );
});

List.displayName = 'List';
List.propTypes = {
  orderable: PropTypes.bool,
  columns: PropTypes.arrayOf(PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.shape({
      id: PropTypes.string,
      title: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.node,
        PropTypes.func,
      ]),
    }),
  ])),
  onOrder: PropTypes.func,
};

export default List;
