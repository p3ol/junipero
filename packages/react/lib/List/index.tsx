import {
  type MutableRefObject,
  type ComponentPropsWithRef,
  forwardRef,
  useImperativeHandle,
  useRef,
  useReducer,
  useCallback,
  useMemo,
} from 'react';
import {
  type ForwardedProps,
  type MockState,
  mockState,
  classNames,
} from '@junipero/core';
import PropTypes from 'prop-types';

import type { ListColumnObject } from '../ListColumn';
import { ListContext, type ListContextType } from '../contexts';
import { ArrowDown, ArrowUp } from '../icons';

export declare type ListRef = {
  orderable: boolean;
  columns: Array<string | ListColumnObject>;
  active?: string | number;
  asc?: boolean | null;
  isJunipero: boolean;
  innerRef: MutableRefObject<any>;
};

export declare interface ListProps extends ComponentPropsWithRef<any> {
  columns?: Array<string | ListColumnObject>;
  onOrder?(order: { column: string | number; asc: boolean | null }): void;
  ref?: MutableRefObject<ListRef | undefined>;
}

export declare interface ListState {
  columns: Array<string | ListColumnObject>;
  active?: string | number;
  asc?: boolean;
}

const List = forwardRef(({
  className,
  children,
  columns = [],
  onOrder,
  ...rest
}: ListProps, ref) => {
  const listRef = useRef([]);
  const innerRef = useRef();
  const orderable = useMemo(() => !!onOrder, [onOrder]);
  const [state, dispatch] = useReducer<MockState<ListState>>(mockState, {
    columns,
    active: null,
    asc: null,
  });

  useImperativeHandle(ref, () => ({
    innerRef,
    orderable,
    columns: state.columns,
    active: state.active,
    asc: state.asc,
    isJunipero: true,
  }));

  const onOrder_ = (column: number, e: MouseEvent) => {
    e?.preventDefault();

    const asc = state.asc === true ? null : state.asc === false;

    dispatch({ active: column, asc });
    onOrder?.({ column, asc });
  };

  const registerColumn = (column: string | ListColumnObject) => {
    if (listRef.current.find(it => it.id === (column as ListColumnObject).id)) {
      return;
    }

    listRef.current.push(column);
    dispatch({ columns: listRef.current });
  };

  const getContext = useCallback<() => ListContextType>(() => ({
    active: state.active,
    asc: state.asc,
    orderable,
    registerColumn,
  }), [
    state.columns,
    state.active,
    state.asc,
    orderable,
  ]);

  const renderColumn = (column:ListColumnObject, index: number) => {
    const { id, title, ...props } = typeof column === 'string'
      ? { id: column, title: column } : column;

    return (
      <th {...props} key={id ?? index}>
        { orderable ? (
          <a href="#" onClick={onOrder_.bind(null, id)}>
            <span className="junipero secondary">{ title }</span>
            { state.active === id && state.asc === true ? (
              <ArrowUp />
            ) : state.active === id && state.asc === false ? (
              <ArrowDown />
            ) : null }
          </a>
        ) : (
          <span className="junipero secondary">{ title }</span>
        ) }
      </th>
    );
  };

  return (
    <ListContext.Provider value={getContext()}>
      <table
        {...rest}
        className={classNames('junipero', 'list', className)}
        ref={innerRef}
      >
        <thead>
          <tr>{ state.columns.map(renderColumn) }</tr>
        </thead>
        <tbody>
          { children }
        </tbody>
      </table>
    </ListContext.Provider>
  );
}) as ForwardedProps<ListProps, ListRef>;

List.displayName = 'List';
List.propTypes = {
  orderable: PropTypes.bool,
  columns: PropTypes.arrayOf(PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.shape({
      id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
      title: PropTypes.oneOfType([
        PropTypes.node,
        PropTypes.string,
      ]),
    }),
  ])),
  onOrder: PropTypes.func,
};

export default List;
