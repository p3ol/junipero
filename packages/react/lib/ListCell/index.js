import { forwardRef, useImperativeHandle, useRef } from 'react';

const ListCell = forwardRef(({
  ...rest
}, ref) => {
  const innerRef = useRef();

  useImperativeHandle(ref, () => ({
    innerRef,
    isJunipero: true,
  }));

  return (
    <td { ...rest } ref={innerRef} />
  );
});

ListCell.displayName = 'ListCell';

export default ListCell;
