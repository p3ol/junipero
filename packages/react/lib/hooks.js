import { useContext } from 'react';

import { FieldControlContext } from './contexts';

export const useFieldControl = () => useContext(FieldControlContext);
