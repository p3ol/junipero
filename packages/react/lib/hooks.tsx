import { useContext } from 'react';

import {
  AlertsContextType,
  AlertsContext,
  DropdownContext,
  DropdownContextType,
  FieldContextType,
  FieldControlContext,
  ListContext,
  ListContextType,
  ToastsContext,
  ToastsContextType,
  ModalContext,
  ModalContextType,
} from './contexts';

export const useFieldControl =
  () => useContext<FieldContextType>(FieldControlContext);
export const useDropdown = () =>
  useContext<DropdownContextType>(DropdownContext);
export const useAlerts = () => useContext<AlertsContextType>(AlertsContext);
export const useList = () => useContext<ListContextType>(ListContext);
export const useToasts = () => useContext<ToastsContextType>(ToastsContext);
export const useModal = () => useContext<ModalContextType>(ModalContext);

export { AlertsContext, DropdownContext, FieldControlContext, ListContext, ToastsContext, ModalContext };
