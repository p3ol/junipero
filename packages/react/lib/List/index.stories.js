import { useState } from 'react';

import List from './';
import ListColumn from '../ListColumn';
import ListItem from '../ListItem';
import ListCell from '../ListCell';

export default { title: 'react/List' };

export const basic = () => (
  <List columns={['name', 'age']}>
    <ListItem item={['John', 25]} />
    <ListItem item={['Jane', 30]} />
  </List>
);

export const verbose = () => (
  <List>
    <ListColumn id="name">Name</ListColumn>
    <ListColumn id="age">Age</ListColumn>

    <ListItem>
      <ListCell>John</ListCell>
      <ListCell>25</ListCell>
    </ListItem>

    <ListItem>
      <ListCell>Jane</ListCell>
      <ListCell>30</ListCell>
    </ListItem>
  </List>
);

export const orderable = () => {
  const [items, setItems] = useState([
    { name: 'John', age: 25 },
    { name: 'Jane', age: 30 },
    { name: 'Jack', age: 20 },
  ]);

  const onOrder = ({ column, asc }) => {
    setItems(it => [...it.sort((a, b) => {
      const aVal = a[column];
      const bVal = b[column];
      const result = aVal > bVal ? 1 : aVal < bVal ? -1 : 0;

      return asc ? result : -result;
    })]);
  };

  return (
    <List onOrder={onOrder}>
      <ListColumn id="name">Name</ListColumn>
      <ListColumn id="age">Age</ListColumn>

      { items.map(item => (
        <ListItem key={item.name} item={[item.name, item.age]} />
      )) }
    </List>
  );
};
