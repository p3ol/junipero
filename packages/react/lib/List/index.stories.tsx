import { useState } from 'react';

import List from '.';
import ListColumn from '../ListColumn';
import ListItem from '../ListItem';
import ListCell from '../ListCell';
import Card from '../Card';

export default { title: 'react/List' };

export const basic = () => (
  <List columns={['name', 'age']}>
    <ListItem item={['John', 25]} />
    <ListItem item={['Jane', 30]} />
  </List>
);

export const verbose = () => (
  <List>
    <ListColumn id="name">Name</ListColumn>
    <ListColumn id="age">Age</ListColumn>

    <ListItem>
      <ListCell>John</ListCell>
      <ListCell>25</ListCell>
    </ListItem>

    <ListItem>
      <ListCell>Jane</ListCell>
      <ListCell>30</ListCell>
    </ListItem>
  </List>
);

export const orderable = () => {
  const [items, setItems] = useState([
    { name: 'John', age: 25 },
    { name: 'Jane', age: 30 },
    { name: 'Jack', age: 20 },
  ]);

  const onOrder = (
    { column, asc }: { column: string | number, asc: boolean}
  ) => {
    setItems(it => [...it.sort((a, b) => {
      const aVal = a[column as 'age' | 'name'];
      const bVal = b[column as 'age' | 'name'];
      const result = aVal > bVal ? 1 : aVal < bVal ? -1 : 0;

      return asc ? result : -result;
    })]);
  };

  return (
    <List onOrder={onOrder}>
      <ListColumn id="name">Name</ListColumn>
      <ListColumn id="age">Age</ListColumn>

      { items.map(item => (
        <ListItem key={item.name} item={[item.name, item.age]} />
      )) }
    </List>
  );
};

export const insideCard = () => (
  <Card className="with-list">
    <List columns={['name', 'age']}>
      <ListItem item={['John', 25]} />
      <ListItem item={['Jane', 30]} />
    </List>
  </Card>
);
