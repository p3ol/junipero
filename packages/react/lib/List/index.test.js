import { useState } from 'react';
import { fireEvent, render } from '@testing-library/react';

import List from './';
import ListItem from '../ListItem';
import ListColumn from '../ListColumn';
import ListCell from '../ListCell';

describe('<List />', () => {
  it('should render using props', () => {
    const { container, unmount } = render(
      <List columns={['name', 'age']}>
        <ListItem item={['John', 25]} />
        <ListItem item={['Jane', 30]} />
      </List>
    );
    expect(container).toMatchSnapshot();
    unmount();
  });

  it('should render using children', () => {
    const { container, unmount } = render(
      <List>
        <ListColumn id="name">Name</ListColumn>
        <ListColumn id="age">Age</ListColumn>

        <ListItem>
          <ListCell>John</ListCell>
          <ListCell>25</ListCell>
        </ListItem>

        <ListItem>
          <ListCell>Jane</ListCell>
          <ListCell>30</ListCell>
        </ListItem>
      </List>
    );
    expect(container).toMatchSnapshot();
    unmount();
  });

  it('should allow to reorder content', () => {
    const Comp = () => {
      const [items, setItems] = useState([
        { name: 'John', age: 25 },
        { name: 'Jane', age: 30 },
        { name: 'Jack', age: 20 },
      ]);

      const onOrder = ({ column, asc }) => {
        setItems(it => [...it.sort((a, b) => {
          const aVal = a[column];
          const bVal = b[column];
          const result = aVal > bVal ? 1 : aVal < bVal ? -1 : 0;

          return asc ? result : -result;
        })]);
      };

      return (
        <List onOrder={onOrder}>
          <ListColumn id="name">Name</ListColumn>
          <ListColumn id="age">Age</ListColumn>

          { items.map(item => (
            <ListItem key={item.name} item={[item.name, item.age]} />
          )) }
        </List>
      );
    };

    const { container, getByText, unmount } = render(<Comp />);
    expect(container).toMatchSnapshot();

    fireEvent.click(getByText('Name'));
    expect(container).toMatchSnapshot();

    fireEvent.click(getByText('Name'));
    expect(container).toMatchSnapshot();

    unmount();
  });
});
