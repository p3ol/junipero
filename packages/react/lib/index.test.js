import { useReducer } from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import { mockState } from '@junipero/core';
import { useTimeout } from '@junipero/hooks';

import TextField from './TextField';

describe('mockState: inside react', () => {
  it('should fail update the state correctly and return the old ' +
    'state', async () => {
    const Comp = () => {
      const [state, dispatch] = useReducer(mockState, {
        value: {
          text: 'This is a test',
          content: [],
        },
      });

      useTimeout(() => {
        dispatch({ value: { ...state.value, content: ['foo'] } });
      }, 1, []);

      const onTextChange = ({ value }) => {
        dispatch({ value: { ...state.value, text: value } });
      };

      const onContentChange = () => {
        dispatch({
          value: { ...state.value, content: [...state.value.content, 'bar'] },
        });
      };

      return (
        <div>
          <TextField
            data-testid="Field"
            value={state.value.text}
            onChange={onTextChange}
          />
          <button onClick={onContentChange}>Add content</button>
          <div>Text: { state.value.text }</div>
          <div>Content: { state.value.content.join(', ') }</div>
        </div>
      );
    };

    const { container, unmount } = render(<Comp />);

    fireEvent.change(screen.getByTestId('Field'),
      { target: { value: 'This is a test 2' } });

    await waitFor(() => screen.getByText('Content: foo'));
    expect(container).toMatchSnapshot('With updated text field value');

    fireEvent.click(screen.getByText('Add content'));

    expect(container)
      .toMatchSnapshot('With updated content but with old text value');

    unmount();
  });

  it('should not fail update the state correctly and return the new ' +
    'state when using mockState with a callback', async () => {
    const Comp = () => {
      const [state, dispatch] = useReducer(mockState, {
        value: {
          text: 'This is a test',
          content: [],
        },
      });

      useTimeout(() => {
        dispatch(s => ({ value: { ...s.value, content: ['foo'] } }));
      }, 1, []);

      const onTextChange = ({ value }) => {
        dispatch(s => ({ value: { ...s.value, text: value } }));
      };

      const onContentChange = () => {
        dispatch(s => ({
          value: { ...s.value, content: [...s.value.content, 'bar'] },
        }));
      };

      return (
        <div>
          <TextField
            data-testid="Field"
            value={state.value.text}
            onChange={onTextChange}
          />
          <button onClick={onContentChange}>Add content</button>
          <div>Text: { state.value.text }</div>
          <div>Content: { state.value.content.join(', ') }</div>
        </div>
      );
    };

    const { container, unmount } = render(<Comp />);

    fireEvent.change(screen.getByTestId('Field'),
      { target: { value: 'This is a test 2' } });

    await waitFor(() => screen.getByText('Content: foo'));
    expect(container).toMatchSnapshot('With updated text field value');

    fireEvent.click(screen.getByText('Add content'));

    expect(container)
      .toMatchSnapshot('With updated content AND updated text field valud');

    unmount();
  });
});
