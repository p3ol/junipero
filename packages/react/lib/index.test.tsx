import { useReducer } from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import { mockState } from '@junipero/core';
import { useTimeout } from '@junipero/hooks';

import type { FieldContent } from './types';
import TextField from './TextField';

interface State {
  value: {
    text: string;
    content: string[];
  };
}

describe('mockState: inside react', () => {
  it('should correctly update the state and return the new ' +
    'state when using mockState with a callback', async () => {
    const Comp = () => {
      const [state, dispatch] = useReducer(mockState<State>, {
        value: {
          text: 'This is a test',
          content: [],
        },
      });

      useTimeout(() => {
        dispatch(s => ({ value: { ...s.value, content: ['foo'] } }));
      }, 1, []);

      const onTextChange = ({ value }: FieldContent) => {
        dispatch(s => ({ value: { ...s.value, text: value } }));
      };

      const onContentChange = () => {
        dispatch(s => ({
          value: { ...s.value, content: [...s.value.content, 'bar'] },
        }));
      };

      return (
        <div>
          <TextField
            data-testid="Field"
            value={state.value.text}
            onChange={onTextChange}
          />
          <button onClick={onContentChange}>Add content</button>
          <div>Text: { state.value.text }</div>
          <div>Content: { state.value.content.join(', ') }</div>
        </div>
      );
    };

    const { container, unmount } = render(<Comp />);

    fireEvent.change(screen.getByTestId('Field'),
      { target: { value: 'This is a test 2' } });

    await waitFor(() => screen.getByText('Content: foo'));
    expect(container).toMatchSnapshot('With updated text field value');

    fireEvent.click(screen.getByText('Add content'));

    expect(container)
      .toMatchSnapshot('With updated content AND updated text field valud');

    unmount();
  });
});
