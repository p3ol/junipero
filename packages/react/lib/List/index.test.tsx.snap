// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> should allow to reorder content 1`] = `
<div>
  <table
    class="junipero list"
  >
    <thead>
      <tr>
        <th>
          <a
            href="#"
          >
            <span
              class="junipero secondary"
            >
              Name
            </span>
          </a>
        </th>
        <th>
          <a
            href="#"
          >
            <span
              class="junipero secondary"
            >
              Age
            </span>
          </a>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          John
        </td>
        <td>
          25
        </td>
      </tr>
      <tr>
        <td>
          Jane
        </td>
        <td>
          30
        </td>
      </tr>
      <tr>
        <td>
          Jack
        </td>
        <td>
          20
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`<List /> should allow to reorder content 2`] = `
<div>
  <table
    class="junipero list"
  >
    <thead>
      <tr>
        <th>
          <a
            href="#"
          >
            <span
              class="junipero secondary"
            >
              Name
            </span>
            <svg
              class="junipero icon arrow-bottom"
              height="5"
              viewBox="0 0 8 5"
              width="8"
            >
              <path
                d="M1 1L4 4L7 1"
              />
            </svg>
          </a>
        </th>
        <th>
          <a
            href="#"
          >
            <span
              class="junipero secondary"
            >
              Age
            </span>
          </a>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          John
        </td>
        <td>
          25
        </td>
      </tr>
      <tr>
        <td>
          Jane
        </td>
        <td>
          30
        </td>
      </tr>
      <tr>
        <td>
          Jack
        </td>
        <td>
          20
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`<List /> should allow to reorder content 3`] = `
<div>
  <table
    class="junipero list"
  >
    <thead>
      <tr>
        <th>
          <a
            href="#"
          >
            <span
              class="junipero secondary"
            >
              Name
            </span>
            <svg
              class="junipero icon arrow-up"
              height="5"
              viewBox="0 0 8 5"
              width="8"
            >
              <path
                d="M1 4L4 1L7 4"
              />
            </svg>
          </a>
        </th>
        <th>
          <a
            href="#"
          >
            <span
              class="junipero secondary"
            >
              Age
            </span>
          </a>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Jack
        </td>
        <td>
          20
        </td>
      </tr>
      <tr>
        <td>
          Jane
        </td>
        <td>
          30
        </td>
      </tr>
      <tr>
        <td>
          John
        </td>
        <td>
          25
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`<List /> should render using children 1`] = `
<div>
  <table
    class="junipero list"
  >
    <thead>
      <tr>
        <th>
          <span
            class="junipero secondary"
          >
            Name
          </span>
        </th>
        <th>
          <span
            class="junipero secondary"
          >
            Age
          </span>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          John
        </td>
        <td>
          25
        </td>
      </tr>
      <tr>
        <td>
          Jane
        </td>
        <td>
          30
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`<List /> should render using props 1`] = `
<div>
  <table
    class="junipero list"
  >
    <thead>
      <tr>
        <th>
          <span
            class="junipero secondary"
          >
            name
          </span>
        </th>
        <th>
          <span
            class="junipero secondary"
          >
            age
          </span>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          John
        </td>
        <td>
          25
        </td>
      </tr>
      <tr>
        <td>
          Jane
        </td>
        <td>
          30
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
